lib/report.js

```diff
@@ -62,8 +62,23 @@ class Report {
       try {
         const path = resolve(this.resolve, v8ScriptCov.url)
         const script = v8toIstanbul(path, this.wrapperLength)
-        script.applyCoverage(v8ScriptCov.functions)
-        map.merge(script.toIstanbul())
+
+        if (v8ScriptCov.functions.every(({ functionName, ranges }) => {
+          if (!script.source.includes(functionName)) {
+            return false
+          }
+
+          for (const { endOffset } of ranges) {
+            if (endOffset > script.source.length) {
+              return false
+            }
+          }
+
+          return true
```

Comment:
https://coveralls.io/builds/23074810/source?filename=lib/report.js#L77

I'm not sure why this line is not covered in Coveralls. If the test doesn't cover this line, L79 and L80 also should not.